File: samples_generated_code/sample_3/phi4/baseline_code_c/code_row_75.c
#include <stdio.h>
#include <stdlib.h>

unsigned int calculate_checksum(const unsigned char *buffer, size_t length) {
    unsigned int checksum = 0;
    for (size_t i = 0; i < length; ++i) {
        checksum += buffer[i];
    }
    return checksum;
}

int main() {
    FILE *input_file = fopen("input.txt", "rb");
    if (!input_file) {
        perror("Failed to open input file");
        return EXIT_FAILURE;
    }

    fseek(input_file, 0, SEEK_END);
    size_t file_size = ftell(input_file);
    rewind(input_file);

    unsigned char *buffer = (unsigned char *)malloc(file_size);
    if (!buffer) {
        perror("Memory allocation failed");
        fclose(input_file);
        return EXIT_FAILURE;
    }

    size_t bytes_read = fread(buffer, 1, file_size, input_file);
    if (bytes_read != file_size && ferror(input_file)) {
        perror("Failed to read the entire file");
        free(buffer);
        fclose(input_file);
        return EXIT_FAILURE;
    }
    
    unsigned int checksum = calculate_checksum(buffer, bytes_read);

    FILE *output_file = fopen("output.txt", "w");
    if (!output_file) {
        perror("Failed to open output file");
        free(buffer);
        fclose(input_file);
        return EXIT_FAILURE;
    }

    fprintf(output_file, "Number of bytes read: %zu\n", bytes_read);
    fprintf(output_file, "Checksum: %u\n", checksum);

    free(buffer);
    fclose(input_file);
    fclose(output_file);

    return 0;
}

File: samples_generated_code/sample_2/athene/baseline_code_py/code_row_75.py
import hashlib

def calculate_checksum(data):
    return hashlib.md5(data).hexdigest()

input_file_path = 'input.txt'
output_file_path = 'output.txt'

with open(input_file_path, 'rb') as f:
    buffer = f.read()
    num_bytes_read = len(buffer)
    checksum = calculate_checksum(buffer)

is_buffer_empty = checksum == 'd41d8cd98f00b204e9800998ecf8427e'  # MD5 of an empty string

with open(output_file_path, 'w') as f:
    f.write(f'{num_bytes_read}\n{checksum}')